rpc_jobs/jjb_setup.yml

```diff
@@ -40,43 +40,52 @@
     dsl: |
       library "rpc-gating@${RPC_GATING_BRANCH}"
       common.internal_slave(){
-        dir("rpc-gating"){
-          stage('Run JJB'){
-            withCredentials([
-              usernamePassword(
-                credentialsId: "service_account_jenkins_api_creds",
-                usernameVariable: "JENKINS_USER",
-                passwordVariable: "JENKINS_API_PASSWORD"
-              )
-            ]){
-              sh """#!/bin/bash -xe
-                source ../.venv/bin/activate
+        try {
+          dir("rpc-gating"){
+            stage('Run JJB'){
+              withCredentials([
+                usernamePassword(
+                  credentialsId: "service_account_jenkins_api_creds",
+                  usernameVariable: "JENKINS_USER",
+                  passwordVariable: "JENKINS_API_PASSWORD"
+                )
+              ]){
+                sh """#!/bin/bash -xe
+                  source ../.venv/bin/activate
 
-                cat > jenkins_jobs.ini << EOF
-      [job-builder]
-      ignore_cache=False
-      keep_descriptions=False
-      include_path=.:scripts:~/git/
-      recursive=True
-      exclude=.*manual:./development
-      allow_deplicates=False
+                  cat > jenkins_jobs.ini << EOF
+        [job-builder]
+        ignore_cache=False
+        keep_descriptions=False
+        include_path=.:scripts:~/git/
+        recursive=True
+        exclude=.*manual:./development
+        allow_deplicates=False
 
-      [jenkins]
-      url=${JENKINS_URL}
-      EOF
+        [jenkins]
+        url=${JENKINS_URL}
+        EOF
 
-                if [[ "$RPC_GATING_BRANCH" == "master" ]] && [[ "$JOBS" == "-r rpc_jobs" ]]; then
-                    echo "Setting --delete-old as RPC_GATING_BRANCH=master and JOBS='-r rpc_jobs'"
-                    UPDATE_ARGS="--delete-old"
-                fi
+                  if [[ "$RPC_GATING_BRANCH" == "master" ]] && [[ "$JOBS" == "-r rpc_jobs" ]]; then
+                      echo "Setting --delete-old as RPC_GATING_BRANCH=master and JOBS='-r rpc_jobs'"
+                      UPDATE_ARGS="--delete-old"
+                  fi
 
-                jenkins-jobs --conf jenkins_jobs.ini \
-                             --user $JENKINS_USER \
-                             --password $JENKINS_API_PASSWORD \
-                             --ignore-cache \
-                             \$JJB_ARGS update \$UPDATE_ARGS $JOBS
-              """
+                  jenkins-jobs --conf jenkins_jobs.ini \
+                               --user $JENKINS_USER \
+                               --password $JENKINS_API_PASSWORD \
+                               --ignore-cache \
+                               \$JJB_ARGS update \$UPDATE_ARGS $JOBS
+                """
+              }
             }
           }
+        } catch(e) {
```

Comment:
This job is frequently run manually when testing a new/modified job, we probably don't want to log the failures from those jobs. I think the if block starting on line 69 is an attempt to identify the builds that were started automatically, maybe the same logic could be used here?

Reply:
Good catch, updated to avoid creating cards when we're running against a specific branch or job.

Reply:
L69 is not about detecting builds that were started automatically. It's for detecting builds that use the master branch and are not limited to a subset of jobs, as those are the conditions required for safely removing old jobs.